api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala

```diff
@@ -232,17 +230,38 @@ abstract class WorksController[M <: MultipleResultsRequest[W],
     * of the form /works/{id}.
     *
     */
-  private def respondWithRedirect(originalUri: String,
+  private def respondWithRedirect(originalRequest: Request,
                                   work: IdentifiedRedirectedWork,
-                                  contextUri: String) =
+                                  contextUri: String) = {
+
+    val path = originalRequest.path.replaceAll(
+      s"/${work.canonicalId}",
+      s"/${work.redirect.canonicalId}"
+    )
+
+    val params = originalRequest.params.toMap
+      .filterNot {
+        case (k,_) => k == "id"
+      }
+      .map {
+        case (k,v) => s"$k=$v"
```

Comment:
Have you checked this handles URL encoding/decoding of parameters in queries correctly?

Reply:
No. It may well not - I'd rather not add more custom code to do URL param conversion as that should definitely be handled by whatever library is parsing these requests. Sadly Finatra does not expose that class (it exists,but is private).